mesonbuild/interpreterbase.py

```diff
@@ -451,6 +454,11 @@ def __init__(self, source_root: str, subdir: str, subproject: str):
         # Current node set during a function call. This can be used as location
         # when printing a warning message during a method call.
         self.current_node = None  # type: mparser.BaseNode
+        # This is set to `version_string` when this statement is evaluated:
+        # meson.version().compare_version(version_string)
+        # If it was part of a if-clause, it is used to temporally override the
+        # current meson version target within that if-block.
+        self.tmp_meson_version = None # type: str
```

Comment:
This should be `T.Optional[str]`
```suggestion
        self.tmp_meson_version = None # type: T.Optional[str]
```